src/main/java/fruitshop/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,18 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        Storage storage = new Storage();
+        int newQuantity = fruitTransaction.getQuantity()
+                + storage.getQuantity(fruitTransaction.getFruit());
+        if (newQuantity >= 0) {
+            storage.set(fruitTransaction.getFruit(), newQuantity);
+        } else {
+            throw new RuntimeException("negative balance cannot be recorded");
+        }
+    }
```

Comment:
The `SupplyOperation` class should not create a new instance of `Storage` in the handler method. Instead, it should work with a single instance of `Storage` that is passed to it, to ensure that all operations are modifying the same underlying storage. Consider revising the design to avoid using a new instance of `Storage` in each operation handler.